R/exclude.R

```diff
@@ -127,9 +127,12 @@ parse_exclusions <- function(file,
 
   if (length(starts) > 0L) {
     if (length(starts) != length(ends)) {
-      starts_msg <- line_info(starts, type = "start")
-      ends_msg <- line_info(ends, type = "end")
-      stop(file, " has ", starts_msg, " but only ", ends_msg, " for exclusion from linting!", call. = FALSE)
+      starts_msg <- line_info(starts, type = "start") # nolint: object_usage_linter.
```

Comment:
All newly introduced nolint directives are due to https://github.com/r-lib/lintr/issues/2252.

Reply:
It seems wise to mark them with the issue number then so they are cleaned up in a PR addressing this issue.

Reply:
I am actually thinking that maybe we should fix #2252 before considering the current PR for a merge.

The current PR will be a good crucible to test it.

WDYT, @MichaelChirico?

Reply:
SGTM, do you want to have a go at implementing?

Reply:
I have instead added TODO linking to this issue to address.

The PR has already been pending for a while, and I don't want to delay its merge even further while trying to find time to work on a different feature.